### PR TITLE
DBZ-6745 Remove redundant filter logic from SignalProcessor

### PR DESCRIPTION
**Pull request by TyrantLucifer**
https://issues.redhat.com/browse/DBZ-6745

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java [hunk 1]
```diff
@@ -69,14 +72,18 @@ public SignalProcessor(Class<? extends SourceConnector> connector,
                            Offsets<P, O> previousOffsets) {
 
         this.connectorConfig = config;
-        this.signalChannelReaders = signalChannelReaders;
+        this.totalChannelReaders = signalChannelReaders;
```
Comment (mfvitale) [inline debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java:75]: WDYT to have a all available channels in the `signalChannelReaders` variable and then introduce a `enabledChannelReades` the will contains the filtered list? I think will be more readable.

File: debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java [hunk 2]
```diff
@@ -69,14 +72,18 @@ public SignalProcessor(Class<? extends SourceConnector> connector,
...
         this.documentReader = documentReader;
         this.previousOffsets = previousOffsets;
         this.signalProcessorExecutor = Threads.newSingleThreadScheduledExecutor(connector, config.getLogicalName(), SignalProcessor.class.getSimpleName(), false);
 
-        signalChannelReaders.stream()
+        // filter single channel reader based on configuration
```
Comment (mfvitale) [inline debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java:80]: I am wondering if it's better to extract a method and avoid the comment.

**Comment by mfvitale** [comment]
Hi @TyrantLucifer Thanks for the improvement, generally LGTM. I have added just few comments. 
Can also please check to the test failures to see if they are related?

File: debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java (hunk 1 above)
Comment (TyrantLucifer) [reply debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java]: Good idea, I will change it.

File: debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java (hunk 2 above)
Comment (TyrantLucifer) [reply debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java:80]: Good idea, I will change it.

**Comment by TyrantLucifer** [comment]
> Hi @TyrantLucifer Thanks for the improvement, generally LGTM. I have added just few comments. Can also please check to the test failures to see if they are related?

I see https://github.com/debezium/debezium/actions/runs/5747822118/job/15580375031 has the same problem that occured in this pull request, I think it maybe the CI workflow unstable. Wait until this round of CI/CD is completed before taking a look. Thank you for your review!

File: debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java [hunk 3]
```diff
@@ -74,9 +77,11 @@ public SignalProcessor(Class<? extends SourceConnector> connector,
         this.previousOffsets = previousOffsets;
         this.signalProcessorExecutor = Threads.newSingleThreadScheduledExecutor(connector, config.getLogicalName(), SignalProcessor.class.getSimpleName(), false);
 
-        signalChannelReaders.stream()
-                .filter(isEnabled())
-                .forEach(signalChannelReader -> signalChannelReader.init(connectorConfig));
+        // filter single channel reader based on configuration
+        this.enabledChannelReader = getEnabledChannelReaders();
```
Comment (mfvitale) [inline debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java:81]: @TyrantLucifer just last one. Can you please rename it to `enabledChannelReaders`?

Comment (TyrantLucifer) [reply debezium-core/src/main/java/io/debezium/pipeline/signal/SignalProcessor.java:81]: Of course, my fault, sorry~

**Review by mfvitale — APPROVED** [review]

**Comment by jpechane** [comment]
@TyrantLucifer Applied, thanks!

**Pull request merged**